spec/classes/init_spec.rb

```diff
@@ -293,6 +293,43 @@
         end
       end
 
+      context 'with log_entries paramter specified as valid array' do
+        let :params do
+          {
+            :log_entries => [ '*.* /var/log/catchall', 'kern.* /var/log/kern.log' ],
+          }
+        end
+        let :facts do
+          {
+            :kernel            => 'Linux',
+            :osfamily          => 'RedHat',
+            :lsbmajdistrelease => '6',
+          }
+        end
+
+        it {
+          should contain_file('rsyslog_config').with_content(/^\*.\* \/var\/log\/catchall\nkern.\* \/var\/log\/kern.log$/)
+        }
+
+      end
+
+      context 'with log_entries specified as an invalid value' do
+        let(:params) { { :log_entries => 'i_am_simply_not_array' } }
+        let :facts do
+          {
+            :kernel            => 'Linux',
+            :osfamily          => 'Debian',
+            :lsbmajdistrelease => '7',
+          }
+        end
+
+        it do
+          expect {
+            should contain_class('rsyslog')
+          }.to raise_error(Puppet::Error)
```

Comment:
Should include the error message. You can find this by putting `Puppet::Error,/asdf/)`, running the spec tests and seeing what the error is.